dist/src/main/java/io/zeebe/broker/StandaloneBroker.java

```diff
@@ -44,32 +72,32 @@ public void run() {
                 }
               }
             });
-    WAITING_LATCH.await();
+    waiting_latch.await();
   }
 
-  private static Broker createBrokerFromConfiguration(final String[] args) {
+  private Broker createBrokerInBaseDirectory() {
     String basePath = System.getProperty("basedir");
 
     if (basePath == null) {
       basePath = Paths.get(".").toAbsolutePath().normalize().toString();
     }
 
-    return new Broker(args[0], basePath, null);
+    return new Broker(configuration, basePath, null);
   }
 
-  private static Broker createDefaultBrokerInTempDirectory() {
+  private Broker createDefaultBrokerInTempDirectory() {
     Loggers.SYSTEM_LOGGER.info("No configuration file specified. Using default configuration.");
 
     try {
       tempFolder = Files.createTempDirectory("zeebe").toAbsolutePath().normalize().toString();
-      final BrokerCfg cfg = new BrokerCfg();
-      return new Broker(cfg, tempFolder, null);
+      final BrokerCfg defaultConfiguration = new BrokerCfg();
```

Comment:
So the previous behavior still allowed environment variables to override the configuration, as resolving was part of the post-processing step. Is there anything that speaks against using the spring provided configuration (which will be default if the user doesn't specify anything)?

Reply:
hmm, very good point. You are correct in your assessment.

I think it comes down to "least astonishment principle". 

My goal here was to say: you either have everything in terms of configuration, or a clean slate with defaults. 

This hybrid to ignore the configuration file, but respect the environment variables is a bit confusing to me. Plus, you would need to make sure that you don't accidentally have environment variables which do define configuration files.

I would yield to your preference in this case, as I don't know how this mode was used in the past.

Reply:
As you noted with the profiles, this is used pretty much only for development. It is useful to be able to configure it if, for example, you want to run a cluster locally through the IDE, but still need to assign different ports.

Reply:
So I would say it should still be configurable through external means, e.g. env var, properties, args, etc

Reply:
In this case I would use the full blown Spring config for both cases. And then these methods would only deal with what folder the broker runs in. Ditto for Gateway.
Please let me know whether this is acceptable for you.

Reply:
Sounds good. The only thing we care about here is that it runs in a temporary directory, really.

Reply:
done